### PR TITLE
#58 Session.userauth_keyboardinteractive method

### PR DESCRIPTION
**Pull request by dvolodin7**
Session.userauth_keyboardinteractive method

### CODE REVIEW

**Comment by pkittenis** [comment]
Hey there,

Thanks for the PR and your contribution!

Can you please remove the libssh2 1.9.0 upgrade as part of this PR. That should be done separately from adding keyboard interactive implementation if you want to make a separate PR for it, or I can.

The kbd interactive changes look good but need a short test to go in.

**Comment by dvolodin7** [comment]
Greetings,

I've cleaned the PR. Please take a look

**Comment by pkittenis** [comment]
Looks good, thanks for making it. Can you add a short test for the new `userauth_keyboardinteractive` session function? See existing [session tests](https://github.com/ParallelSSH/ssh2-python/blob/master/tests/test_session.py).

**Comment by dvolodin7** [comment]
> Looks good, thanks for making it. Can you add a short test for the new `userauth_keyboardinteractive` session function? See existing [session tests](https://github.com/ParallelSSH/ssh2-python/blob/master/tests/test_session.py).

Username and password needed. Embedded OpenSSHServer does not provide one.
I guess it will try agains CI container.

**Comment by pkittenis** [comment]
A failing test is ok, just need to exercise the new code path. It should be raising exception on failure.

**Comment by dvolodin7** [comment]
Tried to. Got AuthenticationError not captured by assertRaises

**Comment by dvolodin7** [comment]
BTW, we're using own custom container for ssh tests - https://code.getnoc.com/infrastructure/tests/testsshd

It allows to build more complicated tests, considering positive and negative scenarios:
https://code.getnoc.com/noc/noc/merge_requests/1990/diffs#0c47f52f2a6192b16db8ccde786bbca459d9c3a2_0_53

File: tests/test_session.py [hunk 1]
```diff
@@ -21,6 +21,18 @@ def test_pubkey_auth(self):
             self.user, self.user_key, publickey=self.user_pub_key,
             passphrase=''), 0)
 
+    def test_failed_password_auth(self):
+        self.assertRaises(
+            AuthenticationError,
+            self.session.userauth_password('FAKE USER', 'FAKE PASSWORD')
```
Comment (pkittenis) [inline tests/test_session.py]: Function and arguments should be passed in separately - `assertRaises` calls the function.

Should be:

```
         self.assertRaises(
            AuthenticationError,
            self.session.userauth_password, 'FAKE USER', 'FAKE PASSWORD'
         )
```

Comment (dvolodin7) [reply tests/test_session.py]: done

**Comment by pkittenis** [comment]
Thanks for that container pointer, looks good. Was looking for something similar - the embedded server can be a bit flaky, and not suited to more complex stuff. Made #81 to migrate to it.
